iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -865,9 +875,9 @@ private async Task SubscribeCloudToDeviceMessagesAsync()
             await _subscribeCompletionSource.Task.ConfigureAwait(true);
         }
 
-        private async Task SubscribeTwinResponsesAsync()
+        private Task SubscribeTwinResponsesAsync()
         {
-            await _channel.WriteAsync(new SubscribePacket(0, new SubscriptionRequest(TwinResponseTopicFilter, QualityOfService.AtMostOnce))).ConfigureAwait(true);
+            return _channel.WriteAsync(new SubscribePacket(0, new SubscriptionRequest(TwinResponseTopicFilter, QualityOfService.AtMostOnce)));
```

Comment:
This used to await the operation and it's not now. how does that change the behavior of the Subscribe method?

Reply:
I see, the caller awaits the task which in turn will await the WriteAsync call.